statsmodels/stats/pscore_match.py

```diff
@@ -0,0 +1,242 @@
+'''Propensity Score Matchig
+
+
+Author: 
+License: BSD-3
+
+'''
+
+import statsmodels.api as sm
+import numpy as np
+from statsmodels.stats.weightstats import CompareMeans, DescrStatsW
+import pandas as pd
+from scipy import stats
+
+
+class PropensityScoreMatch(object):
+    def __init__(self, assigment_index, covariates, treatment_objective_variable,\
+                 algo = 'strata', radius = 0.01, regression = 'logit'):
+        self.assigment_index = assigment_index
+        self.covariates = covariates
+        self.treatment_objective_variable = treatment_objective_variable
+        if regression == 'logit':
+            self.regression_class = sm.Logit
+        elif regression == 'probit':
+            self.regression_class = sm.Probit
+        else:
+            raise
+        if algo == 'strata':
+            self.matching_algo = StrataMatchingAlgorithm(self)
+        elif algo == 'radius':
+            self.matching_algo = RadiusMatchingAlgorithm(self, radius)
+        else:
+            raise
+        self.use_comm_sup = True
+        
+    def fit(self):
+        self.result = PScoreMatchResult()
+        self.compute_pscore()
+        self.matching_algo.fit()
+        
+    def compute_pscore(self):
+        covariates = sm.add_constant(self.covariates, prepend=True)
+        p_mod = self.regression_class(self.assigment_index, covariates)
+        p_res = p_mod.fit(disp=False)
+        #print p_res.summary()
+        self.scores = pd.Series(p_res.predict(covariates), \
+                                index=self.assigment_index.index)
+        self.result.propensity_estimation = p_res
+        if self.use_comm_sup:
+            comm_sup = self.common_support()
+            scores = self.scores[comm_sup]
+            #print 'Using common support: %s' % str([scores.min(), scores.max()]) 
+        else:
+            scores = self.scores
+        #print 'propensity scores description'
+        #print scores.describe()
+        
+        
+    def treatment_effect(self):
+        return self.matching_algo.treatment_effect()
+        
+    def treatment_variance(self):
+        return self.matching_algo.treatment_variance()
+        
+    def treatment_std(self):
+        return np.sqrt(self.treatment_variance())
+        
+    def confidence_interval(self, alpha=0.05):
+        #assuming normality (don't know df)
+        mean, std = self.treatment_effect(), self.treatment_std()
+        return stats.norm.interval(1-alpha, loc=mean, scale=std )
+        
+        
+    def _basic_treated(self):
```

Comment:
Our classes are usually optimized to run once for efficiency, so we don't have a lot of accessor methods like this.  If needed, we have a decorator called cache_readonly that allows methods to behave like attributes.